kurtosis-devnet/pkg/deploy/template.go

```diff
@@ -121,6 +175,10 @@ func (f *Templater) Render() (*bytes.Buffer, error) {
 		f.buildJobs = make(map[string]*dockerBuildJob)
 	}
 
+	if f.buildWg == nil {
```

Comment:
that feels a bit confusing, as I don't think we should be reusing an existing WaitGroup here.


What about this instead?
- remove the buildWg from Templater
- create a WaitGroup unconditionally here
- pass it explicitly to the localPrestateOption and localContractArtifactsOption (so they don't have to check if buildWg is nil, in particular)
- actually share the WaitGroup with dockerBuilds logic below

That would parallelize everything, and probably  make the integration simpler